packages/suite/src/actions/suite/metadataActions.ts

```diff
@@ -541,13 +532,14 @@ export const init = (force = false) => async (dispatch: Dispatch, getState: GetS
     // 2. set device metadata key (master key). Sometimes, if state is not present
     if (
         device.metadata.status === 'disabled' ||
-        (device.metadata.status === 'cancelled' && force)
+        (device.metadata.status === 'cancelled' && force && device?.connected)
     ) {
         dispatch({ type: METADATA.SET_INITIATING, payload: true });
         await dispatch(setDeviceMetadataKey());
     }
 
-    // did user confirm labeling on device?
+    // did user confirm labeling on device? or maybe device was not connected
+    // so suite does not have keys and needs to stop here
     if (getState().suite.device?.metadata.status !== 'enabled') {
```

Comment:
`device` was destructed on L522, `getState().suite` can be removed.

Reply:
I don't think you are right on this. I need to get updated value as there is another async action preceding this. One would think reference to object once declared somewhere up would be enough, but it is not. So that code is there on purpose actually.